### DISCUSSION TITLE
You need a better explanation of what typia is/does

### DISCUSSION THREAD

**Discussion by axefrog** [samchon/typia#1116] [Ideas]
On your website's landing page:

- "Only one line" -- Only one line of what? To do what?

- "No extra schema required" -- Extra schema? What extra schema? We need a schema for something? What is typia? What does it have to do with schemas?

- "Just fine with pure TypeScript type" -- Oookay? I still have no idea what typia is, so I don't know how this benefit relating to TypeScript types is relevant to anything.

- "`typia.assert<T>(input);`" -- I can see that you're making a point about conciseness here. Conciseness is good. But I still don't know what typia is or how it helps me, so demonstrating how concise it is is premature.

Then, further below:

- "Super-fast Runtime Validator" -- Speed is good. So I guess typia is faster than whatever it's competing against? But I still don't know what typia is. Is it a validator, or does it _have_ a validator? What's it validating? Is validation the main feature, or is it just something that's an important part of what it does? _What is typia?_

- "Fast JSON Serialization" -- Again, same thing. I don't know what typia does, so the fact that it has fast JSON serialization isn't helping me judge its efficacy in addressing my needs.

Anyway you get the picture. I get the impression that your landing page has started with an assumption that someone arriving there is already aware of your product and is looking for a solution that they know typia could possibly help with.

But what about people who don't know what typia is? I am a developer in the broader AI space, and I was recently sent a link (among many I frequently receive) to the typia homepage, but I found myself frustrated because I felt like I was looking at something that could be useful to me, and yet even after reading through the whole landing page, I still had no idea what was being offered. My suggestion is to drop all assumptions about who you're talking to, and pretend that the only thing you know is that you're talking to a developer and that they probably know what TypeScript is. But that's it. Therefore, your landing page should start with something short and to the point that expresses what problem you're solving and who it's for. Make sure that by the time I reach the feature list, I already know whether or not typia is relevant to me, even if I have never done any development in the space that typia is targeting.

**Comment by samchon** [comment]
Not easy mission \o/. 

To accomplish this issue, I have to entirely remake the intro page.

May need lots of time to start.

**Comment by ryoppippi** [comment]
@samchon 
How about enable discussion feature for this Repo and move this issue there.

**Comment by ryoppippi** [comment]
So this is not updated and this looks spam.
We can delete this unproductive comment

**Comment by axefrog** [comment]
I'm still curious what this is for. "Not easy mission" ? I don't buy that for a second. Just imagine you're in the real world talking to a developer who is very experienced in many areas, and they ask you what your project is for. You hand them your phone on which you've loaded up your web page as it is right now. They read the text you've written, and then they look at you confused and say "man, I've been developing with TypeScript since the early days and have worked on projects of many different kinds, but I have no idea what your project does." They then say to you "I'll give you $10,000 right now if you can explain to me who your project is for and what problem it helps them with". Would you just say "not easy mission" and then walk away without the $10,000? Or would you try to think about how it might be possible that another developer might not immediately fit all the assumptions that your web page is making about them?

Different developers have expertise on different areas of development, right? Would you assume that someone who is an expert in networking and distributed data synchronisation is also an expert in web framework development? Of course you wouldn't.

So let's say we have two developers; one for each of the two areas of specialisation I just mentioned. The networking guy releases a library that offers some really useful tools related to networking, and these tools would actually help the web framework guy with some of the things their web framework has to think about on the server side. Problem is, the networking guy thinks that all developers are networking guys, and so his web page presents the library's value proposition in a way that feels like it's only intended to be read by experienced networking specialists. It explains things in terms that essentially amount to "Hey, you know how you're a networking guy like me? I know that you are familiar with a bunch of things that we networking guys deal with every day, so I'm not even gonna bother mentioning those things. So anyway, this library solves problems NETWORKING_A and NETWORKING_B, and boy, don't we all hate that problem!" Of course only a networking guy would be familiar with problems NETWORKING_A and NETWORKING_B. To anyone else reading the page, they don't understand what the networking guy is so excited about. The assumptions the web page is making about the reader don't actually hold true unless the reader is a networking specialist, and so anyone else reading the page will lack the context necessary to see why the library is useful.

So now the web framework guy comes along and reads the page about this networking library, and keeps seeing references to things that make no sense to him, and gets a sense that it is failing to offer critical information that would provide wider context for anyone who isn't a networking guy! See, the networking guy wants his library to be useful to all developers who have any need to deal with networking concerns, but is only bothering to make an effort to convince other networking guys that his library is worth using. Stop assuming everyone is a networking guy. You're not only selling your product to an echo chamber of other networking guys. You're selling it to anyone who would benefit from the problems it solves.

So now back to your developer friend looking at your phone out in the real world. You know they are experienced, highly competent, and would benefit from what your library does. But now you also know that they might lack a bit of context regarding things that feel like common knowledge to you, but which actually are only common knowledge for developers with your particular specialisation. Your friend isn't an idiot. You're not talking to a junior developer here. They'll get what you mean if you simply offer sufficient context. What would you tell them to explain why your library is useful? What context do they lack? How are you gonna make sure you walk away with that $10,000?

To wrap things up, I would suggest that you should take what I'm saying seriously as a career skill. Being able to explain what you're doing to other developers, being able to write good documentation, and more generally, being effective at communicating, are essential skills for any top-tier developer. So treat this as a challenge, not a chore.

(Note: English as a second language is not an issue here. This is about general communication, not about language or grammar.)

**Reply by samchon** [reply]
Understand the importance, but as I only develop `typia` like open source project in my spare time, it is not a easy mission for me. 

According to your view, shouldn't the intro page of typia be entirely rewritten? I am not unaware of the importance of documentation and wording, but since this is an open source project for hobby purposes and not a commercial project, I cannot aim for perfection in everything. Furthermore, since there are countless functions to be developed, such a website publishing cannot have higher priority for me.

Of course, if you help me, it's a completely different story. Even though I've started the typia project only by myself, but it has become a medium-sized project used by quite a many people. I hope that the many people participate and contribute to this typia project. Can I expect a nice PR from you? I'm not very confident with documentation, design/publishing, wording, etc., so I'd appreciate some help.

**Reply by axefrog** [reply]
I get where you're coming from, and appreciate your concerns. I would suggest that next time you have a spare 30 minutes, just try writing one paragraph that introduces Typia to an experienced developer who is not already familiar with your niche. I'm sure if you're spending a lot of your spare time on this project that it has come up in conversation when you talk to other developers in real life (or at least online, e.g. on Discord). So if nothing else, it'd be a good opportunity to think about how you respond when someone (who is a developer) says "so tell me about this project you're working on". And if the paragraph feels good, maybe you just paste it up the top of your existing landing page and then leave the rest of that page mostly unchanged.

Seeing as I spent all this time annoying you about this, I'd be happy to give you feedback if you do write anything. Let me know! Good luck with it all, in any case.

**Comment by samchon** [comment]
Updated
